homeassistant/components/modbus/binary_sensor.py

```diff
@@ -73,6 +74,7 @@ async def async_setup_slaves(
         # this ensures that idx = bit position of value in result
         # polling is done with the base class
         name = self._attr_name if self._attr_name else "modbus_sensor"
+        assert name is not DEVICE_CLASS_NAME
```

Comment:
This does not look right, “assert” in production code ! you should log an error.

apart from that modbus uses a generated name, so I think the assert will always be called

Reply:
The purpose is to make the type checker happy + catch future attempts to use `DEVICE_CLASS_NAME` which is incompatible with the way modbus generates names by string manipulation.

We have plenty of asserts in production code already, so I don't think it's a problem.
I can add explaining comments to all asserts added in this PR, both in modbus and elsewhere?

Reply:
Adding a comment would solve my concern.